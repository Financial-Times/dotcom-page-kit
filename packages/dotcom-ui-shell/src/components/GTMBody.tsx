import React from 'react'
import { TFlagsData } from '@financial-times/dotcom-ui-flags/src/types'

// This component is maintained by the ads team
const GTMBody = ({ flags }: { flags: TFlagsData }) => {
  if (!flags.enableGTM) {
    return null
  }

  return (
    <noscript>
      <iframe
        src="https://www.googletagmanager.com/ns.html?id=GTM-NWQJW68"
        height="0"
        width="0"
        style={{
          display: 'none',
          visibility: 'hidden'
        }}
      />
    </noscript>
  )
}

GTMBody.defaultProps = {
  flags: {}
}

export default GTMBody
