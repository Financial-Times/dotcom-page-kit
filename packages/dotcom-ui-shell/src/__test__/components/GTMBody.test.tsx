import React from 'react'
import renderer from 'react-test-renderer'
import GTMBody from '../../components/GTMBody'

describe('dotcom-ui-shell/src/components/GTMBody', () => {
  it('renders the gtm body script when the enableGTM flag is on', () => {
    const props = {
      flags: {
        enableGTM: true
      }
    }

    const tree = renderer.create(<GTMBody {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders null when the enableGTM flag is off', () => {
    const props = {
      flags: {
        enableGTM: false
      }
    }

    const tree = renderer.create(<GTMBody {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders null when there is no enableGTM', () => {
    const props = {
      flags: {}
    }

    const tree = renderer.create(<GTMBody {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
