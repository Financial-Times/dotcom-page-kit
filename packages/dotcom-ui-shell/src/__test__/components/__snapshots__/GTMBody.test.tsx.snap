// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dotcom-ui-shell/src/components/GTMBody renders null when the enableGTM flag is off 1`] = `null`;

exports[`dotcom-ui-shell/src/components/GTMBody renders null when there is no enableGTM 1`] = `null`;

exports[`dotcom-ui-shell/src/components/GTMBody renders the gtm body script when the enableGTM flag is on 1`] = `
<noscript>
  <iframe
    height="0"
    src="https://www.googletagmanager.com/ns.html?id=GTM-NWQJW68"
    style={
      {
        "display": "none",
        "visibility": "hidden",
      }
    }
    width="0"
  />
</noscript>
`;
